google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeployAction.java

```diff
@@ -167,7 +155,20 @@ public void onTextAvailable(ProcessEvent event, Key outputType) {
     public void processTerminated(final ProcessEvent event) {
       try {
         if (event.getExitCode() == 0) {
-          callback.succeeded(new DeploymentRuntimeImpl(deploymentOutput.toString(), version));
+
+          // Parse JSON output to retrieve service/version info.
+          DeployOutput deployOutput = null;
+          try {
+            deployOutput = parseDeployOutput(deploymentOutput.toString());
+          } catch (JsonParseException e) {
+            getLoggingHandler().print(
```

Comment:
you can use consoleLogLn here instead. Also, did we settle on not being notified on this kind of failure (I don't think we get a report on logger.warn).  Otherwise, lgtm.